### ISSUE TITLE
Using more than one android resources library/directory, R not found

### ISSUE THREAD

**Issue by Nikolas-LFDesigns** [bazelbuild/rules_kotlin#787]
In our project we use flat resources structure like

```
package:
- ResourceUser.java
- resources_common
- res
```
and BUILD.bazel like this:
```
android_library(
    name = "resource_user",
    resource_files = glob(["res/**"]),
    srcs = ["ResourceUser.java"],
    deps = [
        ":resources_common",
        [...]
    ],
)

android_library(
    name = "resources_common",
    resource_files = glob(["resources_common/**"]),
    [...]
)
```
One even could depend on some common resources from other packages if appropriate.

When trying to replace android_library definition in resource_user from example above with kotlin rules' kt_android_library (even not actually converting anything to kotlin at all and leaving a java file) and building the project, the kotlin compiler seems confused and can't find resource_files' R definitions that are linked in the [resource_user] library (but can find resources from the [resources_common] android library which is in its deps).

An error is as follows:  
```
ERROR: [project_root]/kotlin-android-resources-example/example/src/main/java/examples/android/lib/BUILD.bazel:3:19: KotlinCompile //example/src/main/java/examples/android/lib:example_fragment_kt { kt: 1, java: 0, srcjars: 0 } for k8 failed: (Exit 1): build failed: error executing command bazel-out/host/bin/external/io_bazel_rules_kotlin/src/main/kotlin/build '--flagfile=bazel-out/k8-fastbuild/bin/example/src/main/java/examples/android/lib/example_fragment_kt-kt.jar-0.params'
example/src/main/java/examples/android/lib/MainFragment.kt:16:35: error: unresolved reference: layout
        return inflater.inflate(R.layout.sample_layout, container,  /*attachToRoot=*/false)
                                  ^
example/src/main/java/examples/android/lib/MainFragment.kt:20:39: error: unresolved reference: id
        view.findViewById<TextView>(R.id.text_view).text = getString(R.string.hello_new)
                                      ^
Jun 29, 2022 12:38:07 AM worker request 0
```
Might be related to some internal resource library not properly linked, but .params-0 .params-1 files for an example_fragment_kt-kt look A-OK to me (at least they have all the resources listed here). 

**Rules kotlin:** 1.6.0
**Bazel:** 4.1.0, reproducible even on 5.2.0

**Example which illustrates the case:**  
https://github.com/Nikolas-LFDesigns/kotlin-android-resources-example

**Similar issues:**
An issue https://github.com/bazelbuild/bazel/issues/8109 sounds similar to me, but it was fixed eons ago (still, might be useful on triaging)

**Comment by arunkumar9t2** [comment]
The root cause is due to `AndroidManifest.xml`. Since you are using same manifest file for both targets, two R class will be generated with same package name in the classpath and only one will get imported. In your case, it seems like `R` class from `resources_second` is getting imported which does not contain layout files.

You could fix it by either specifying different package name via `custom_package` or use a separate `AndroidManifest.xml` with a different package name.

This worked:

```diff
 android_library(
     name = "resources_second",
     manifest = "AndroidManifest.xml",
+    custom_package = "examples.android.lib.res",
     resource_files = glob(["res-second/**"]),
     visibility = ["//visibility:public"],
 )
```

**Comment by Nikolas-LFDesigns** [comment]
The point is, we **need** two libraries with the same R class package and standard Bazel rules (android_library, or even android_binary) handle it gracefully by merging resources from different libraries on the same physical package as needed. That way, we can construct a library of any complexity and atomicity as we would imagine.
Also don’t forget that way we can use common resources from package1 in package2 without sharing a whole “src/main/res”, like ex. Gradle does by default.

**Comment by arunkumar9t2** [comment]
I just realized this is a supported case in bazel. The root problem still remains, i.e two different R class with the same package name in the classpath. Bazel handles it by ensuring the merged R jar comes [first](https://github.com/bazelbuild/bazel/blob/525df32b5dc60448e1abcd3d537ed3454a841946/src/main/java/com/google/devtools/build/lib/rules/android/AndroidCommon.java#L512-L513) so that alone gets imported. Attempted fix in #790

**Comment by Nikolas-LFDesigns** [comment]
Great job!
Though provided solution might use some generalization.. @cgruber @Bencodes  could you help us here?
